### PR TITLE
Blog app - Processing data

### PR DESCRIPTION
**Pull request by Donard97**
**I created this PR to check if my project is as requested.** 👀 👀

_In this project I set up models for tables of Users, Posts, Comments, and Likes._ 💯 💯 

### Project requirements that are done:
 

- Created model classes for all entities as shown in the ERD diagram. ☑️ 

- Methods that are implemented on models : 

1. Users: 
A method that returns the 3 most recent posts for a given user. ☑️ 

2. Posts:
A method that updates the posts counter for a user. ☑️ 
A method that returns the 5 most recent comments for a given post. ☑️ 

3. Comments:
 A method that updates the comments counter for a post. ☑️ 

4. Likes:
  A method that updates the likes counter for a post. ☑️ 

- Created a method current_user in my ApplicationController that returns the first user from a database ☑️ 

<hr>

_Feel free to review 👀  my project and let me know if there are any essential changes 💪_

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user, foreign_key: 'user_id'
```
Comment (rloterh) [inline app/models/comment.rb]: - [x] As per the given _ERD_, the _Comment_ **association** should have _Comment_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'` 

    Kindly ensure to improve upon this aspect.

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :user, foreign_key: 'user_id'
```
Comment (rloterh) [inline app/models/like.rb]: - [x] As per the given _ERD_, the _Like_ **association** should have _Like_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'` 
   
   Kindly ensure to improve upon this aspect.

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user, foreign_key: 'user_id'
```
Comment (rloterh) [inline app/models/post.rb]: - [x] As per the given _ERD_, the _Post_ **association** should have _Post_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'` 

    Kindly ensure to improve upon this aspect.

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'user_id'
+  has_many :comments, foreign_key: 'user_id'
+  has_many :likes, foreign_key: 'user_id'
```
Comment (rloterh) [inline app/models/user.rb]: - [x] As per the given _ERD_, the _User_ **association** should have _**author_id**_ (_should be different from the name **user**_) referenced as **foreign key** in _Post_, _Comment,_ and _Like_ models. Kindly ensure to improve this aspect, as can be seen below:
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Donard, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦  🚦  

## Good Points 👍  
- Descriptive Pull Request
- Good ReadMe
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb (hunk 1 above)
Comment (Donard97) [reply app/models/comment.rb]: Thank you for letting me know. I did it.

File: app/models/like.rb (hunk 2 above)
Comment (Donard97) [reply app/models/like.rb]: Thank you for letting me know. I did it.

File: app/models/post.rb (hunk 3 above)
Comment (Donard97) [reply app/models/post.rb]: Thank you for letting me know. I did it.

File: app/models/user.rb (hunk 4 above)
Comment (Donard97) [reply app/models/user.rb]: Thank you for letting me know. I did it.

**Review by Emmyn5600 — APPROVED** [review]
STATUS: APPROVED :+1: 

Hello, @Donard97  👋🏾 👋🏾 congratulations on implementing all the changes requested by the previous reviewer as well as the project requirements.

## Optional suggestions

- N/a

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

**Pull request merged**